packages/fxa-metrics-processor/src/lib/pubsub.ts

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Service, Inject } from 'typedi';
+import { v1, Message, PubSub } from '@google-cloud/pubsub';
+import * as Sentry from '@sentry/node';
+import { EventEmitter, once } from 'events';
+
+import Config from '../config';
+import { MessageProcessor } from '../lib/message-processor';
+
+const queueConfig = Config.getProperties().queues;
+
+@Service()
+export class QueueProcessor {
+  private subClient: any;
+  private pubSubClient: PubSub;
+  private formattedRawEventSubscription: any;
+  private shouldStop = false;
+  private emitter = new EventEmitter();
+  private sentry = Sentry;
+
+  @Inject()
+  private messageProcessor!: MessageProcessor;
+
+  constructor() {
+    this.subClient = new v1.SubscriberClient();
+    this.pubSubClient = new PubSub({ projectId: queueConfig.pubsubProjectId });
+    this.formattedRawEventSubscription = this.subClient.subscriptionPath(
+      queueConfig.pubsubProjectId,
+      queueConfig.rawEvents
+    );
+  }
+
+  /**
+   * Retrieve a given batch of messages from the raw event queue.
```

Comment:
I see an `await` not wrapped in a try-catch. Is it possible for this function to throw errors? What happens if the client connection fails? Would be good to document expected behavior in the doc block.